x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -206,29 +192,8 @@ private static Object innerDeepCopy(Object value, boolean unmodifiable) {
         }
     }
 
-    private static class CacheKey {
-
-        final String enrichIndex;
-        final SearchRequest searchRequest;
-
-        private CacheKey(String enrichIndex, SearchRequest searchRequest) {
-            this.enrichIndex = enrichIndex;
-            this.searchRequest = searchRequest;
-        }
-
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            CacheKey cacheKey = (CacheKey) o;
-            return enrichIndex.equals(cacheKey.enrichIndex) && searchRequest.equals(cacheKey.searchRequest);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(enrichIndex, searchRequest);
-        }
-    }
+    // Visibility for testing
+    record CacheKey(String enrichIndex, Object value, int maxMatches) {}
```

Comment:
Very minor point: It threw me off at first that the main field in my key is `value` -- why does my key contain my value (it doesn't)?. I think that this field is only named `value` because we pull the key out of the ingest document as the value of another field in AbstractEnrichProcessor. It might be helpful to future maintainers to call this `keyData` or something like that.

Reply:
++! I'm inclined to add line breaks and a per-member comment/justification of what they are and why they're there.